src/anonymous-credentials/schema.ts

```diff
@@ -587,56 +624,78 @@ export class CredentialSchema extends Versioned {
     return `dk:${term}`;
   }
 
-  static flattenJSONSchema(node: any) {
-    if (typeof node.type !== 'string') {
-      throw new Error(`Schema node must have type field that is a string, got ${JSON.stringify(node)}`);
-    }
+  /**
+   * Convert a schema object as per JSON-schema syntax (`IJsonSchema`) to the internal representation (`ISchema`).
+   * Currently, does not check if the needed JSON-schema definitions are actually present but assumes that they will be
+   * already passed.
+   * @param node
+   * @param nodeKeyName - Name of the node, used for throwing more informative error message
+   */
+  static convertToInternalSchemaObj(node: any, nodeKeyName: string = ''): object {
```

Comment:
since we care about micro optimisations (assuming because you're using set over array here now)... why are we converting to an internal type then storing it? this class then has to have two member objects: internal schema and json schema, which are only set at the constructor. why? this is a waste of memory when previously we just stored one non mutated jsonschema and then parsed/flattened when required

Reply:
I explained above that the impl. of flattening is more efficient and flattening is used in both credentials and presentations. So we tradeoff memory for CPU time.

Reply:
its not worth any more changes over this as its negligable